runtime/compiler/control/HookedByTheJit.cpp

```diff
@@ -628,6 +690,21 @@ static void jitHookInitializeSendTarget(J9HookInterface * * hook, UDATA eventNum
             }
 #endif // defined(J9VM_INTERP_AOT_COMPILE_SUPPORT) && defined(J9VM_OPT_SHARED_CLASSES) && (defined(TR_HOST_X86) || defined(TR_HOST_POWER) || defined(TR_HOST_S390) || defined(TR_HOST_ARM) || defined(TR_HOST_ARM64))
          } // if (TR::Options::sharedClassCache())
+
+#if defined(J9VM_OPT_JITSERVER)
+      // If the method is cached at the server side, set the count to the lower number
+      // regardless of whether the count has been set before
+      if (methodCachedAtServer)
```

Comment:
I think we should also test `TR::Options::getCountsAreProvidedByUser()`.
Let's say that the use specified -Xaot:count=300, then we should use that 300 value regardless of whether the method is cached at the server or not.
Then we can simplify `calculateCountForMethodCachedAtServer()` to be just `optionsAOT->getInitialSCount()`